plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -206,3 +213,13 @@ def sweep(self, arguments: List[str]) -> None:
         )
         log.info(f"Best parameters: {best_trial.params}")
         log.info(f"Best value: {best_trial.value}")
+
+    def get_sampler(self) -> BaseSampler:
+        if self.optuna_config.sampler == Sampler.tpe:
+            return TPESampler(seed=self.optuna_config.seed)
+        if self.optuna_config.sampler == Sampler.random:
+            return RandomSampler(seed=self.optuna_config.seed)
+        if self.optuna_config.sampler == Sampler.cmaes:
+            return CmaEsSampler(seed=self.optuna_config.seed)
+
+        raise NotImplementedError(f"{self.optuna_config.sampler} is not supported.")
```

Comment:
You can still have a dictionary with sampler classes in the config or in the code.

```python
samplers = {
  "tpe" : "optuna.samplers.TPESampler",
  ...
}
if self.optuna_config.sampler not in samplers:
  raise ...

sampler_class = samplers[self.optuna_config.sampler]
sampler = sampler_class(seed=self.optuna_config.seed)
```

Reply:
Thank you for your suggestion. I used the dictionary in the code. I think it resolved the test failure of `test_optuna_sweeper_plugin.py::test_launch_jobs`. Please refer to this commit https://github.com/facebookresearch/hydra/pull/1132/commits/816a4a970e6f2e25ed3fd5e8729aca40319c0371.